packages/astro/test/test-utils.js

```diff
@@ -70,6 +70,13 @@ export async function loadFixture(inlineConfig) {
 	if (!inlineConfig || !inlineConfig.root)
 		throw new Error("Must provide { root: './fixtures/...' }");
 
+	// Compatible with different Node versions (https://vitejs.dev/guide/migration.html#dev-server-changes)
+	if (!inlineConfig?.server) {
+		inlineConfig.server = {
+			host: '127.0.0.1'
+		}
+	}
```

Comment:
I think we should fix the test to support a different host instead of setting a specific host here. What error are you getting that's causing the issue?

Reply:
I don't think this is resolved. I was suggesting fixing the failing tests to also accept both ipv4 and ipv6 so we can test both cases.

Reply:
@bluwy I've been unable to replicate with running `mocha --timeout 20000 **/html-page.test.js` manually, both on Node v14.18.0 and Node 16.16.0. Also [suggested some debugging options here.](https://github.com/withastro/astro/issues/5254#issuecomment-1299109253) I personally wouldn't be comfortable merging this fix if neither of us can replicate the issue

Reply:
Ah my bad @bholmesdev I missed that message. It's an issue with Node 18 (or >= 17 specifically) due to the default [dns resolve order](https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder) change. It was `ipv4first` before, but node >=17 onwards, it's `verbatim`.

I actually also just ran and found A LOT of test errors, so I kinda understand why this PR went with a simple fix 😅  I guess this review would be fine then @wulinsheng123.

Though I'll try to find another alternative so we don't test ipv4 only.

Reply:
Oh boy 😅 thanks for clarifying! Agreed this test is pretty minimal and wouldn't mind merging

Reply:
> Ah my bad @bholmesdev I missed that message. It's an issue with Node 18 (or >= 17 specifically) due to the default [dns resolve order](https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder) change. It was `ipv4first` before, but node >=17 onwards, it's `verbatim`.
> 
> I actually also just ran and found A LOT of test errors, so I kinda understand why this PR went with a simple fix 😅 I guess this review would be fine then @wulinsheng123.
> 
> Though I'll try to find another alternative so we don't test ipv4 only.

Thanks your this review,

Reply:
> I don't think this is resolved. I was suggesting fixing the failing tests to also accept both ipv4 and ipv6 so we can test both cases.

I agreed, But this Error only happens in `test-utils.js` and you even specify '127.0.0.1' to resolve

`const resolveUrl = (url) =>`http://${'127.0.0.1'}:${config.server.port}${url.replace(/^\/?/, '/')}`;`

Reply:
Yeah we'll have to update the test util to support both ipv4 and verbatim too in the future. For now let's get this workaround in so it can be tested in node 18.

Adding a TODO:
```suggestion
	// TODO: Remove this to test in Node >= 17 where the dns resolver is verbatim
	if (!inlineConfig?.server) {
		inlineConfig.server = {
			host: '127.0.0.1'
		}
	}
```

Reply:
Ok no problem let me try late

Reply:
I'm guessing it would be changing 

https://github.com/withastro/astro/blob/e7dc8b956d7b54b585c4469da258e1d5c051cee9/packages/astro/test/test-utils.js#L121

to `localhost` instead of `127.0.0.1`. But I'm pretty sure that's going to cause some other issues, it might need some work to identify how to fix that onwards, which I'm not sure yet 😅